testgen/generators/collation_short.py

```diff
@@ -98,52 +232,49 @@ def generateCollTestData2(self, filename, icu_version, start_count=0):
 
         root_locale = re.compile("@ root")
         locale_string = re.compile("@ locale (\S+)")
-        test_line = re.compile("^\*\* test:(.*)")
-        rule_header_pattern = re.compile("^@ rules")
-        rule_pattern = re.compile("^&.*")
-        strength_pattern = re.compile("% strength=(\S)")
+        test_line = re.compile("^\*\* test:is_attribute(.*)")
         compare_pattern = re.compile("^\* compare(.*)")
 
         comparison_pattern = re.compile(
-            "(\S+)\s+(\S+)\s*(\#?.*)"
+            "(\S+)\s+(\S+)\s*(#?.*)"
         )  # compare operator followed by string
 
-        attribute_test = re.compile("^\% (\S+)\s*=\s*(\S+)")
         rules = ""
-        strength = None
+        case_first = None
+        case_level = None
+        numeric = None
+        backwards = None
 
         # Ignore comment lines
         string1 = ""
         string2 = ""
-        attributes = []
+        attributes = {}
         test_description = ""
 
-        # Get @ root or @ locale ...
-        # Check for "@ rules"
-        # Handle % options, e.g., strengt=h, reorder=, backwards=, caseFirst=,
-        #  ...
-        # Find "* compare" section and create list of tests for this,
-        # starting comparison with empty string ''.
-        # Handle compre options =, <, <1, <2, <3, <4
-
         locale = ""
-        line_number = 0
+        line_number = 0  # Starts at one for actual lines
         num_lines = len(raw_testdata_list)
         while line_number < num_lines:
+            # line_number += 1
             line_in = raw_testdata_list[line_number]
-            line_number += 1
 
             is_comment = recommentline.match(line_in)
             if line_in[0:1] == "#" or is_comment or reblankline.match(line_in):
+                line_number += 1
                 continue
 
             if root_locale.match(line_in):
```

Comment:
Can you add a line explaining that seeing a new locale means creating a new collator instance?